### PR TITLE
ADD: change doc for release 2.5.0 and 2.4.3

### PR DESCRIPTION
**Pull request by xuanyu66**
### What problem does this PR solve? 

Add changelog for 2.5.0 and 2.4.3

### Check List 

Tests 

 - No code

Related changes

 - Need to update the documentation

- Need to cherry-pick to release-2.4

### CODE REVIEW

**Comment by xuanyu66** [comment]
@shiyuhang0 @zhangyangyu @sunxiaoguang PTAL

File: README.md [hunk 1]
```diff
@@ -53,7 +52,7 @@ You can download the nightly version (master branch) [here](https://download.pin
 
 ## How to build from sources
 
-TiSpark now supports Spark 2.3.0+、2.4.0+ and 3.0.0+. The earlier TiSpark versions for Spark 2.1.0+ only contain bug-fixes. After these versions, you can still get support for Spark 2.1 until TiSpark 1.2.1.
+TiSpark now supports Spark 2.3.0+、2.4.0+、3.0.0+ and 3.1.0+. The earlier TiSpark versions for Spark 2.1.0+ was not supported anymore.
```
Comment (zhangyangyu) [inline README.md:55]:
```suggestion
TiSpark now supports Spark 2.3.0+、2.4.0+、3.0.0+ and 3.1.0+. The earlier TiSpark versions for Spark 2.1.0+ are not supported anymore. 
```

File: README.md [hunk 2]
```diff
@@ -13,14 +13,14 @@ Read the [Quick Start](./docs/userguide.md).
 
 ## Getting TiSpark
 
-+ Currently, TiSpark 2.3.16 is the latest stable version, which is highly recommended. It is compatible with Spark 2.3.0+ and Spark 2.4.0+. It is also compatible with TiDB-2.x、TiDB-3.x and TiDB-4.x. Please follow the [document](./docs/userguide.md).
++ Currently, TiSpark 2.4.3, 2.5.0 is the latest stable version, which is highly recommended. It is compatible with Spark 2.3.0+ and Spark 2.4.0+. It is also compatible with TiDB-2.x、TiDB-3.x and TiDB-4.x. Please follow the [document](./docs/userguide.md).
```
Comment (zhangyangyu) [inline README.md:16]: We don't support TiDB-2.x and TiDB-3.x right?

Comment (xuanyu66) [inline README.md:16]:
```suggestion
+ Currently, TiSpark 2.4.3, 2.5.0 is the latest stable version, which is highly recommended. Please follow the [document](./docs/userguide.md).
```

File: README.md [hunk 3]
```diff
@@ -13,14 +13,14 @@ Read the [Quick Start](./docs/userguide.md).
...
 
-+ If you are using TiDB-5.0 and Spark 2.3.0+/2.4.0+, TiSpark 2.4.0 is recommended.
++ If you are using TiDB-5.0 and Spark 2.3.0+/2.4.0+, TiSpark 2.4.3 is recommended.
 
-+ If you are using Spark 3.0, TiSpark 2.5.0 is recommended.
++ If you are using Spark 3.0/3.1, TiSpark 2.5.0 is recommended.
   
-+ TiSpark 1.2.1 is the latest stable version compatible with Spark 2.1.0+.
-  - When using TiSpark 1.2.1, follow the [document for Spark 2.1](./docs/userguide_spark2.1.md).
++ TiSpark 1.2.1 is compatible with Spark 2.1.0+, which was not supported anymore
```
Comment (xuanyu66) [inline README.md:22]:
```suggestion
+ TiSpark 1.2.1 is compatible with Spark 2.1.0+, which is not supported anymore 
```

File: README.md (hunk 2 above)
Comment (xuanyu66) [reply README.md]: fixed

File: README.md [hunk 4]
```diff
@@ -13,14 +13,14 @@ Read the [Quick Start](./docs/userguide.md).
 
 ## Getting TiSpark
 
-+ Currently, TiSpark 2.3.16 is the latest stable version, which is highly recommended. It is compatible with Spark 2.3.0+ and Spark 2.4.0+. It is also compatible with TiDB-2.x、TiDB-3.x and TiDB-4.x. Please follow the [document](./docs/userguide.md).
++ Currently, TiSpark 2.4.3, 2.5.0 is the latest stable version, which is highly recommended. Please follow the [document](./docs/userguide.md).
```
Comment (xuanyu66) [inline README.md:16]:
```suggestion
+ Currently, TiSpark 2.4.3, 2.5.0 is the latest stable version, which is highly recommended. You can get compatibility information from [table](## How to choose TiSpark Version) .Please follow the [document](./docs/userguide.md).
```

File: README.md [hunk 5]
```diff
@@ -13,14 +13,14 @@ Read the [Quick Start](./docs/userguide.md).
 
 ## Getting TiSpark
 
-+ Currently, TiSpark 2.3.16 is the latest stable version, which is highly recommended. It is compatible with Spark 2.3.0+ and Spark 2.4.0+. It is also compatible with TiDB-2.x、TiDB-3.x and TiDB-4.x. Please follow the [document](./docs/userguide.md).
++ Currently, TiSpark 2.4.3, 2.5.0 is the latest stable version, which is highly recommended. You can get compatibility information from [table](##How to choose TiSpark Version) .Please follow the [document](./docs/userguide.md).
```
Comment (zhangyangyu) [inline README.md:16]: The link does not work :-)

**Comment by shiyuhang0** [comment]
https://github.com/pingcap/tispark/pull/2101 this pr

File: README.md (hunk 5 above)
Comment (xuanyu66) [inline README.md:16]:
```suggestion
+ Currently, TiSpark 2.4.3, 2.5.0 is the latest stable version, which is highly recommended. You can get compatibility information from [Markdown - table](##How to choose TiSpark Version) .Please follow the [document](./docs/userguide.md).
```

Comment (xuanyu66) [reply README.md]: fixed

**Review by zhangyangyu — APPROVED** [review]

**Comment by xuanyu66** [comment]
@sunxiaoguang Plz add label `need-to-cherry-pick-to-2.4`

**Comment by shiyuhang0** [comment]
> > #2101 this pr is important for fix the bacthwrite data loss problem. should we put it into changelog
> 
> @shiyuhang0 It had already been released in version https://github.com/pingcap/tispark/releases/tag/v2.4.2
but release-2.4 misses in changelog, add them in this pr or another?

**Pull request merged**